subprojects/tooling-api/src/test/groovy/org/gradle/tooling/internal/adapter/WeakIdentityHashMapTest.groovy

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.internal.adapter
+
+import spock.lang.Specification
+
+
+class WeakIdentityHashMapTest extends Specification {
+
+    private def runGc() {
+        System.gc()
+        Thread.sleep(2000)
```

Comment:
🤔 I'm not completely satisfied with always sleeping for 2 seconds for every test. I agree that reusing `ConcurrentTestUtil` here might be too much so how about introducing something like a `waitForConditionAfterGC(Closure condition)` function here, with a loop to sleep _at most_ 2 seconds but that could return earlier?